src/main/java/org/prebid/server/hooks/execution/v1/HookDecorator.java

```diff
@@ -0,0 +1,20 @@
+package org.prebid.server.hooks.execution.v1;
+
+import org.prebid.server.hooks.v1.Hook;
+import org.prebid.server.hooks.v1.InvocationContext;
+
+import java.util.Objects;
+
+public abstract class HookDecorator<PAYLOAD, CONTEXT extends InvocationContext> implements Hook<PAYLOAD, CONTEXT> {
+
+    protected final Hook<PAYLOAD, CONTEXT> hook;
+
+    public HookDecorator(Hook<PAYLOAD, CONTEXT> hook) {
+        this.hook = Objects.requireNonNull(hook);
+    }
+
+    @Override
+    public String code() {
+        return hook.code();
+    }
+}
```

Comment:
Another one. I get the point in making hierarchy 'idiomatic', but this still looks like an overkill to me. Please, remove this.